components/content-service/pkg/storage/gcloud.go

```diff
@@ -361,46 +399,49 @@ func (rs *DirectGCPStorage) Upload(ctx context.Context, source string, name stri
 	uploadSpan.SetTag("bucket", bucket)
 	uploadSpan.SetTag("obj", object)
 
+	err = gcpEnsureExists(ctx, rs.client, bucket, rs.GCPConfig)
+	if err != nil {
+		err = xerrors.Errorf("unexpected error: %w", err)
+		return
+	}
+
 	var firstBackup bool
 	if _, e := obj.Attrs(ctx); e == gcpstorage.ErrObjectNotExist {
 		firstBackup = true
 	}
 
 	var wg sync.WaitGroup
-	var written int64
 
 	wg.Add(1)
 
 	go func() {
 		defer wg.Done()
 
-		wc := obj.NewWriter(ctx)
-		wc.Metadata = options.Annotations
-		wc.ContentType = options.ContentType
-		// Increase chunk size for faster uploading
-		wc.ChunkSize = googleapi.DefaultUploadChunkSize * 4
-
-		written, err = io.Copy(wc, sfn)
-		if err != nil {
-			log.WithError(err).WithField("name", name).Error("Error while uploading file")
-			return
+		sa := ""
+		if rs.GCPConfig.CredentialsFile != "" {
+			sa = fmt.Sprintf(`-o "Credentials:gs_service_key_file=%v"`, rs.GCPConfig.CredentialsFile)
 		}
 
-		// persist changes in GCS
-		err = wc.Close()
+		args := fmt.Sprintf(`gsutil -q -m %v\
+		  -o "GSUtil:parallel_composite_upload_threshold=150M" \
+		  -o "GSUtil:parallel_process_count=3" \
+		  -o "GSUtil:parallel_thread_count=6" \
```

Comment:
Six threads seem to request lots of CPU resources, I captured the below picture from `gen51` cluster.
![image](https://user-images.githubusercontent.com/49380831/177091016-b76ad1b1-545e-493a-aa9e-4dda2f875678.png)

Should we keep the parameter as now _or_ should we limit more CPU resources for gsutil?

Reply:
> Should we keep the parameter as now or should we limit more CPU resources for gsutil?

We have already limited the number of simultaneous backups to three. Also, the trade-off we are doing here is less RAM and better transfer speeds with more CPU utilization (short period of time).

That said if you have an idea to improve this, please post it here

Reply:
I tried to use https://github.com/EmbarkStudios/gsutil. That should consume fewer resources, but I could get it working consistently.